fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricLanguageProvider.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.provider;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import net.minecraft.data.DataProvider;
+import net.minecraft.data.DataWriter;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
+
+/**
+ * Extend this class and implement {@link FabricLanguageProvider#generateLanguages(LanguageConsumer)}.
+ * Make sure to use {@link FabricLanguageProvider#FabricLanguageProvider(FabricDataGenerator, String, Path)}  FabricLanguageProvider} to declare what language code is being generated if it isn't en_us
+ *
+ * <p>Register an instance of the class with {@link FabricDataGenerator#addProvider} in a {@link net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint}
+ */
+public abstract class FabricLanguageProvider implements DataProvider {
+	protected final FabricDataGenerator dataGenerator;
+	private final String languageCode;
+	private final Path existingLanguageFile;
+
+	protected FabricLanguageProvider(FabricDataGenerator dataGenerator, @Nullable Path existingLanguageFile) {
+		this.dataGenerator = dataGenerator;
+		this.languageCode = "en_us";
+		this.existingLanguageFile = existingLanguageFile;
```

Comment:
```suggestion
		this(dataGenerator, "en_us", existingLanguageFile);
```

Reply:
Maybe also provide one that calls `this(dataGenerator, null)` and another that calls `this(dataGenerator, languageCode, null)`? Or is that too much?

Reply:
okay new commit makes this moot